java/core/src/java/org/apache/orc/Reader.java

```diff
@@ -390,8 +399,15 @@ public SearchArgument getSearchArgument() {
       return preFilterColumns;
     }
 
+    /**
+     * Get the search argument leaves' column-name mapping.
+     * @return an array of column names, one for every predicate leave, or NULL when search argument not set
+     */
     public String[] getColumnNames() {
-      return columnNames;
+      if (this.sarg != null) {
+        return sarg.getLeaves().stream().map(l -> l.getColumnName()).toArray(String[]::new);
```

Comment:
Since this logic doesn't look consistent with the old usage, I have a few questions.
- What if the user-given `columeNames` has a different order from the leaves?
- What if the user-given `columnNames` are different case-sensitively?
- What if the user-given `columnNames` has more columns or less columns?

cc @omalley , too.

Reply:
Hey @dongjoon-hyun these are valid concerns, in general columnNames var is not used anymore (even on 1.5 branch).
The actual mapping to internal colIdx is happening as part of mapSargColumnsToOrcInternalColIdx which in turn calls RecordReaderImpl.findColumns where:

* Depending on the schema isSchemaEvolutionCaseAware var -- the SArgs columnNames can be case-insensitive or not
* If there is not reader schema-file type mapping the colName will be -1 (no Sargs applied for that column)
* If the column is not found in the Reader schema at all, it will throw an exception

In that sense, its the caller's responsibility to use the right Reader schema, with correct column names, define if needed the case-sensitivity (as part of schema.isSchemaEvolutionCaseAware) and handle any missing schema type issues.